scilpy/reconst/fodf.py

```diff
@@ -0,0 +1,130 @@
+    for quantitative comparisons with Sharpening Deconvolution Transform (SDT).
+
+    Parameters
+    ----------
+    data: ndarray
+        4D Input diffusion volume with shape (X, Y, Z, N)
+    bvals: ndarray
+        1D bvals array with shape (N,)
+    bvecs: ndarray
+        2D (normalized) bvecs array with shape (N, 3)
+    full_frf: ndarray
+        frf data, ex, loaded from a frf_file, with shape (4,).
+    sh_order: int, optional
+        SH order used for the CSD. (Default: 8)
+    nbr_processes: int, optional
+        Number of sub processes to start. Default = none, i.e use the cpu count.
+    mask: ndarray, optional
+        3D mask with shape (X,Y,Z)
+        Binary mask. Only the data inside the mask will be used for
+        computations and reconstruction. Useful if no white matter mask is
+        available.
+    sh_basis: str, optional
+        Spherical harmonics basis used for the SH coefficients.Must be either
+        'descoteaux07' or 'tournier07' (default 'descoteaux07')
+        - 'descoteaux07': SH basis from the Descoteaux et al. MRM 2007 paper
+        - 'tournier07': SH basis from the Tournier et al. NeuroImage 2007 paper.
+    return_sh: bool, optional
+        If true, returns the sh.
+    n_peaks: int, optional
+        Nb of peaks for the fodf. Default: copied dipy's default, i.e. 5.
+    force_b0_threshold: bool, optional
+        If True, will continue even if the minimum bvalue is suspiciously high.
+
+    Returns
+    -------
+    peaks_csd: PeaksAndMetrics
+        An object with ``gfa``, ``peak_directions``, ``peak_values``,
+        ``peak_indices``, ``odf``, ``shm_coeffs`` as attributes
+    """
+
+    # Checking data and sh_order
+    check_b0_threshold(force_b0_threshold, bvals.min())
+    if data.shape[-1] < (sh_order + 1) * (sh_order + 2) / 2:
+        warnings.warn(
+            'We recommend having at least {} unique DWI volumes, but you '
+            'currently have {} volumes. Try lowering the parameter sh_order '
+            'in case of non convergence.'.format(
+                (sh_order + 1) * (sh_order + 2) / 2, data.shape[-1]))
+
+    # Checking bvals, bvecs values and loading gtab
+    if not is_normalized_bvecs(bvecs):
+        logging.warning('Your b-vectors do not seem normalized...')
+        bvecs = normalize_bvecs(bvecs)
+    gtab = gradient_table(bvals, bvecs, b0_threshold=bvals.min())
+
+    # Checking full_frf and separating it
+    if not full_frf.shape[0] == 4:
+        raise ValueError('FRF file did not contain 4 elements. '
+                         'Invalid or deprecated FRF format')
+    frf = full_frf[0:3]
+    mean_b0_val = full_frf[3]
+
+    # Checking if we will use parallel processing
+    parallel = True
+    if nbr_processes is not None:
+        if nbr_processes <= 0:
+            nbr_processes = None        # C'est ok ça?? parallel reste true si nbr_processes <0? Je laisse comme c'était.
```

Comment:
Please write in English.

Reply:
I just forgot to erase it. I actually wanted to ask you. Why do you deal with a negative nbr_processes? Why do you keep parallel=true in that case?

Reply:
@jchoude ?